src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -141,7 +141,10 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 DefType templateType = GenericTypesTemplateMap.GetActualTemplateTypeForType(factory, _type.ConvertToCanonForm(CanonicalFormKind.Specific));
 
                 if (templateType.IsCanonicalSubtype(CanonicalFormKind.Any) && !factory.NecessaryTypeSymbol(templateType).RepresentsIndirectionCell)
+                {
                     dependencyList.Add(factory.NativeLayout.TemplateTypeLayout(templateType), "Template Type Layout");
+                    dependencyList.Add(factory.ConstructedTypeSymbol(templateType), "Template type");
```

Comment:
I want to bring attention to this line. I don't know who should be guaranteeing the presence of the constructed EEType for the template type. Right now, nobody guarantees it and at most we get "necessary". This was fine when "necessary" meant "constructed", but that's not the case anymore. I assume on the Project X side this was compensated by the fact that NUTC does the analysis.